contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -537,22 +542,36 @@ def _find_timecode_track_start(track):
     aaf_metadata = track.metadata.get("AAF", {})
 
     # Is this a Timecode track?
-    if aaf_metadata.get("MediaKind") == "Timecode":
-        edit_rate = aaf_metadata.get("EditRate", "0")
-        fps = aaf_metadata.get("Segment", {}).get("FPS", 0)
-        start = aaf_metadata.get("Segment", {}).get("Start", "0")
-
-        # Often times there are several timecode tracks, so
-        # we use a heuristic to only pay attention to Timecode
-        # tracks with a FPS that matches the edit rate.
-        if edit_rate == str(fps):
-            return otio.opentime.RationalTime(
-                value=int(start),
-                rate=float(edit_rate)
-            )
+    if aaf_metadata.get("MediaKind") not in {"Timecode", "LegacyTimecode"}:
+        return
+
+    # Edit Protocol section 3.6 specifies PhysicalTrackNumber of 1 as the
+    # Primary timecode
+    try:
+        physical_track_number = aaf_metadata["PhysicalTrackNumber"]
+    except KeyError:
+        raise AAFAdapterError("Timecode missing 'PhysicalTrackNumber'")
+
+    if physical_track_number != 1:
+        return
```

Comment:
Just so I understand, there can be multiple timecode tracks with physical track number != 1 and they're not what we're looking for, but *if* there is one with the number 1 then its the edit rate for the track?

Reply:
More or less. I wouldn't say it's the edit rate for the track, it's that it's the primary timecode offset.

I think this addresses the idea that in production cases you could have multiple timecodes associated with something. For instance when working with telecine dailies, you have the tape timecode you've captured (usually starting at a specific hour offset based on shoot day) but there would also be a film timecode which is expressed by a numbering system burned into the edge of the film when it was manufactured. For the post-production use case the tape timecode is wheat you work with most the time. Then at the very end of the process you'd want to use that secondary timecode to map back to the original negative.

The language from the AAF spec says:

> A top-level composition shall contain one or more timecode tracks and include a Primary timecode track.

And then a table below shows that "Primary" will have a Physical track number of 1 and alternative timecodes will have other physical track numbers. My interpretation of this is that there will always be a "Primary" and that it is most likely the timecode a user would be interested in.

Reply:
Is it possible to have more than one with a physical track number of 1?  Or does the spec prohibit this?

Reply:
The text is a little ambigous but it is strongly implied that this is the case. Anecdotally from some aaf examples I saw it does seem to be the case.

Reply:
Sounds good to me.  Should be obvious if we ever hit this case.